pkg/hub/client/public/filter_test.go

```diff
@@ -159,9 +160,127 @@ func TestImplementationRequirementsSatisfiedByFilters(t *testing.T) {
 			},
 			requirementsSatisfiedBy: []*gqlpublicapi.TypeInstanceValue{
 				{
-					TypeRef: &gqlpublicapi.TypeReferenceWithOptionalRevision{
+					TypeRef: &gqlpublicapi.TypeReferenceInput{
 						Path:     "cap.type.gcp.sa",
-						Revision: ptr.String("0.1.1"),
+						Revision: "0.1.1",
+					},
+				},
+			},
+		},
+		{
```

Comment:
What about such scenario:
```go
{
			name: "Return Implementations satisfied by injected RequiredTypeInstances",
			expRevision: []gqlpublicapi.ImplementationRevision{
				fixImplementationRevision("without-any-requirements", "0.0.1"),
				fixImplementationRevisionWithRequire("with-gcp-sa-requirement", "0.1.0", gqlpublicapi.ImplementationRequirement{
					AllOf: []*gqlpublicapi.ImplementationRequirementItem{
						{
							Alias: ptr.String("gcp-sa"),
							TypeRef: &gqlpublicapi.TypeReference{
								Path:     "cap.type.gcp.sa",
								Revision: "0.1.1",
							},
						},
						{
							TypeRef: &gqlpublicapi.TypeReference{
								Path:     "cap.core.type.platform.kubernetes",
								Revision: "0.1.1",
							},
						},
					},
				}),
			},
			requiredTIInjectionSatisfiedBy: []*gqlpublicapi.TypeInstanceValue{
				{
					TypeRef: &gqlpublicapi.TypeReferenceInput{
						Path:     "cap.type.gcp.sa",
						Revision: "0.1.1",
					},
				},
			},
			requirementsSatisfiedBy: []*gqlpublicapi.TypeInstanceValue{
				{
					TypeRef: &gqlpublicapi.TypeReferenceInput{
						Path:     "cap.core.type.platform.kubernetes",
						Revision: "0.1.1",
					},
				},
			},
		},
```

I understand that it currently it's additional level but maybe it should be also complementary. So if we see sth like that 
- we simple add `requiredTIInjectionSatisfiedBy` into `requirementsSatisfiedBy` to pass our check, 
- Or change the implementation but this will change also API: (not 100% sure about that approach)
```go
for _, item := range implReq {
		if item.TypeRef == nil {
			continue
		}

		// required TypeInstance injection
		if item.Alias != nil {
			if !contains(requiredTIInjectionSatisfiedBy, *item.TypeRef) {
				// injection needed and TypeInstance for injection not found
				return false
			}
		} else {
			if !contains(availableReq, *item.TypeRef) {
				// all needs to be satisfied, so we can already give up
				return false
			}
		}
	}
```

Reply:
As discussed offline, I will merge all provided `requiredTIInjectionSatisfiedBy` into `requirementsSatisfiedBy` to cover the following use cases:
- the one listed above 
- using `requiredTIInjectionSatisfiedBy` without `requirementsSatisfiedBy`
Also, I will add unit tests for these two cases.